pkg/controller/traceflow/controller_test.go

```diff
@@ -101,7 +101,7 @@ func TestTraceflow(t *testing.T) {
 
 		tfc.kubeClient.CoreV1().Pods("ns1").Create(context.TODO(), &pod1, metav1.CreateOptions{})
 		tfc.client.CrdV1alpha1().Traceflows().Create(context.TODO(), &tf1, metav1.CreateOptions{})
-		res, _ := tfc.waitForTraceflow("tf1", crdv1alpha1.Running, time.Second)
+		res, _ := tfc.waitForTraceflow("tf1", crdv1alpha1.Running, time.Second*8)
```

Comment:
I don't think it is about wait time when updating the traceflow. The failure shows that traceflow controller fails to find the target Pod when processing the Traceflow add request. So I think you should add wait or precheck on the existence of the target Pod before adding TF object in line 103

Reply:
Thanks, corrected and re-run unit test.